src/main/groovy/wooga/gradle/githubReleaseNotes/GithubReleaseNotesPlugin.groovy

```diff
@@ -31,26 +28,20 @@ class GithubReleaseNotesPlugin implements Plugin<Project> {
     static final String EXTENSION_NAME = "GithubReleaseNotes"
     static final String GENERATE_RELEASE_NOTES_TASK = "generateReleaseNotes"
 
-    static Logger logger = Logging.getLogger(GithubReleaseNotesPlugin)
-
     @Override
     void apply(Project project) {
         project.pluginManager.apply(GithubBasePlugin)
         createDefaultTasks(project)
 
-        project.tasks.withType(GenerateReleaseNotes, new Action<GenerateReleaseNotes>() {
-            @Override
-            void execute(GenerateReleaseNotes task) {
-                def defaultStrategy = new DefaultGeneratorStrategy()
-                task.strategy.set(defaultStrategy)
-                task.releaseName.set(project.provider({project.version.toString()}))
-                task.output.set(new File(project.buildDir,"outputs/release-notes.txt"))
-                //task.output.set(project.layout.buildDirectory.dir("outputs").map({ it.file("release-notes.txt") }))
-            }
-        })
+        project.tasks.withType(GenerateReleaseNotes) { GenerateReleaseNotes task ->
+            def defaultStrategy = new DefaultGeneratorStrategy()
+            task.strategy.convention(defaultStrategy)
+            task.releaseName.convention(project.provider({project.version.toString()}))
+            task.output.convention(project.layout.buildDirectory.file("outputs/release-notes.txt"))
+        }
```

Comment:
The tasks avoidance changes should also be along with this PR or another one?

Reply:
The changes amounts for just this block, so I figured that there was no harm in putting it here. But if you think otherwise, I can put them in another PR, no problem.

Reply:
Now I just figured out that this isn't using configuration avoidance, because we are using the `withType(Class, Closure)` signature. I should had used the `withType(Class).configureEach(Closure)` structure if I was to use configuration avoidance. So, I'll leave this happy mistake this way, and we make the configuration avoidance changes in another branch, what do you think?

Reply:
Sounds good to me